runtime/runtime/src/lib.rs

```diff
@@ -283,138 +339,172 @@ impl Runtime {
         debug!(target: "runtime", "{}", log_str);
     }
 
-    fn parallel_validate_transactions(
-        config: &RuntimeConfig,
+    /// Processes a group of transactions (same signer_id and public_key), sorted by ascending nonce.
+    /// Validates, calculates cost, and performs checks sequentially.
+    /// If a transaction fails validation or verification, it invokes handle_invalid_transaction
+    /// and, if `RelaxedChunkValidation` is enabled, skips to the next transaction without failing
+    /// the whole group. The resulting state (`final_state`) contains `Account` and `AccessKey` after
+    /// the last successful transaction in the group to be committed to the `state_update`.
+    /// Returns a `BatchResult` containing verified transactions, final state, and signer_id.
+    fn process_grouped_transactions(
+        &self,
+        group: TransactionBatch,
+        apply_state: &ApplyState,
+        state_update: &TrieUpdate,
         gas_price: Balance,
-        signed_txs: impl IntoParallelIterator<Item = SignedTransaction>,
+        block_height: BlockHeight,
         current_protocol_version: ProtocolVersion,
-    ) -> Vec<(CryptoHash, Result<(ValidatedTransaction, TransactionCost), InvalidTxError>)> {
-        signed_txs
-            .into_par_iter()
-            .map(|signed_tx| {
-                (
-                    signed_tx.get_hash(),
-                    match validate_transaction(config, signed_tx, current_protocol_version) {
-                        Ok(validated_tx) => {
-                            match tx_cost(
-                                config,
-                                &validated_tx,
-                                gas_price,
-                                current_protocol_version,
-                            ) {
-                                Ok(cost) => Ok((validated_tx, cost)),
-                                Err(e) => Err(InvalidTxError::from(e)),
-                            }
-                        }
-                        Err((e, _tx)) => Err(e),
-                    },
-                )
-            })
-            .collect()
+        mut temp_state: Option<(Account, AccessKey)>,
+    ) -> Result<BatchResult, RuntimeError> {
+        let first_idx = group.indices[0];
```

Comment:
```suggestion
        let Some(first_idx) = group.indices.first() else { return ... };
```

There is no reason to risk the node aborting here if somebody passed in an empty batch for some reason, especially since empty input logically translates to an empty output.